src/app/meds-n-labels/activity-3/activity-3.component.ts

```diff
@@ -39,18 +40,22 @@ export class DoctorchatComponent implements OnInit, AfterViewChecked {
     }
 
     send(sentMessage) {
-        this.chatMessage = '';
-        if (sentMessage) {
-            this.messages.push({message: sentMessage, status: 'sent-message'});
-            if (localStorage.getItem(DoctorchatComponent._localStorageKey)) {
-                this._doctorchat.doctorMessage({message: sentMessage}).subscribe(response => {
-                    if (response.reply.hasOwnProperty('link') && response.reply.hasOwnProperty('image')) {
-                      this.messages.push({message: response.reply.message, status: 'recv-message', link: response.reply.link, image: response.reply.image});
-                    } else {
-                      this.messages.push({message: response.reply, status: 'recv-message'});
-                    }
-                });
-            }
+      this.chatMessage = '';
+      if (sentMessage) {
+        this.messages.push({message: {text: sentMessage}, status: 'sent-message'});
+        if (localStorage.getItem(DoctorchatComponent._localStorageKey)) {
+          this._botService.getIntent(sentMessage).subscribe(intent => {
+            this._botService.getResponse(sentMessage).subscribe(text => {
```

Comment:
Is there a default timeout.. or some error handling done here ?

Reply:
Ya, done in the service code.